### PR TITLE
EncodedPrivateKey: migrate from NetworkParameters to Network

### PR DESCRIPTION
**Pull request by schildbach**

### CODE REVIEW

File: core/src/main/java/org/bitcoinj/core/EncodedPrivateKey.java [hunk 1]
```diff
@@ -26,31 +28,46 @@
...
+    }
+
     /**
      * @return network this data is valid for
+     * @deprecated Use {@link #network()}
      */
+    @Deprecated
     public final NetworkParameters getParameters() {
-        return params;
+        return NetworkParameters.of(network);
```
Comment (msgilligan) [inline core/src/main/java/org/bitcoinj/core/EncodedPrivateKey.java:58]: This adds an indirect dependency on `Networks.get()`

**Review by msgilligan — APPROVED** [review]
I approve this PR as it follows our general pattern of moving from `NetworkParameters` to `Network`.

I had proposed PR #2626 which tried to move away from using any idea of "network" in `EncodedPrivateKey` (instead storing the `version` byte) and potentially making it easier for us to replace the dependency on `Networks.get()`, but those issues can be dealt with after this PR is merged.

File: core/src/main/java/org/bitcoinj/core/EncodedPrivateKey.java (hunk 1 above)
Comment (schildbach) [reply core/src/main/java/org/bitcoinj/core/EncodedPrivateKey.java:58]: Yes, but it's deprecated so it will go away in the next release.

**Comment by schildbach** [comment]
Merged.

**Pull request merged**